File: leetcode/dynamic/best_time_to_stock.py
def maxProfit(self, prices: List[int]) -> int:
    # Kadane's Algorithm
    # Same problem as Max Subarray
    maxPrice = 0
    minPrice = prices[0]
    for i in prices:
        if i < minPrice:
            minPrice = i
        elif i - minPrice > maxPrice:
            maxPrice = i - minPrice
    return maxPrice

File: leetcode/dynamic/best_time_to_stock.cpp
int maxProfit(vector<int>& prices) {
	int minPrice = prices[0];
	int maxProfit = 0;
	
	for (int p : prices) {
		if (p < minPrice) {
			minPrice = p;
		} else if (p - minPrice > maxProfit) {
			maxProfit = p - minPrice;
		}
	}
	return maxProfit;
}
